lib/spack/spack/util/environment.py

```diff
@@ -24,6 +24,32 @@
 ##############################################################################
 import os
 
+system_paths = ['/', '/usr/', '/usr/local']
+suffixes = ['lib', 'lib64', 'include']
+system_dirs = sum(([os.path.join(p, s)
+                    for s in suffixes] for p in system_paths), [])
+system_bins = [os.path.join(p, 'bin') for p in system_paths]
+
+
+def filter_system_paths(paths):
+    return [p for p in paths if not any(p in s for s in system_dirs)]
+
+
+def filter_system_bin_paths(paths):
+    filtered = []
+    bins = []
+    # filter out system bin paths
+    for p in paths:
+        if p in system_bins:
+            bins.append(p)
+        else:
+            filtered.append(p)
+
+    # add to the begining because PATH is constructed with prepend_path(item)
+    filtered[:0] = bins
```

Comment:
My fault : 
```python
filtered = [x for x in dedupe(bins + paths)]
```
I made it a generator to avoid constructing a list if the temporary was big.